src/test/scala/com/fulcrumgenomics/sv/tools/SvPileupTest.scala

```diff
@@ -361,4 +393,76 @@ class SvPileupTest extends UnitSpec {
 
     SvPileup.filterTemplate(template, minPrimaryMapq=30, minSupplementaryMapq=20) shouldBe None
   }
+
+  /** Writes the given BAM records to a BAM file */
+  private def toInput(rec: SamRecord*): PathToBam = {
+    val path    = makeTempFile("input.", ".bam")
+    val header  = new SamBuilder().header
+    val writer  = SamWriter(path, header, sort=None)
+    writer ++= rec
+    writer.close()
+    path
+  }
+
+  private object Outputs {
+    def apply(): Outputs = {
+      val prefix = makeTempFile("output", "")
+      Outputs(
+        prefix = prefix,
+        bam    = PathUtil.pathTo(prefix + ".bam"),
+        txt    = PathUtil.pathTo(prefix + ".bam")
+      )
+    }
+  }
+
+  private case class Outputs(prefix: PathPrefix, bam: PathToBam, txt: FilePath) {
+    Seq(prefix, bam, txt)foreach(_.toFile.deleteOnExit())
+  }
+
+  "SvPileup" should "run end to end" in {
+    // A set of reads where each read (r1/r2) is either completely aligned near one side of
+    // breakpoint, or the read is split-read aligned across the breakpoint.
+    val fullR1  = r("chr1",   1, Plus,  r=1, cigar="100M",   supp=false)
+    val r1Half1 = r("chr1", 100, Plus,  r=1, cigar="50M50S", supp=false)
+    val r1Half2 = r("chr7", 800, Plus,  r=1, cigar="50S50M", supp=true)
+
+    val fullR2  = r("chr7", 850, Minus, r=2, cigar="100M",   supp=false)
+    val r2Half1 = r("chr7", 800, Minus, r=2, cigar="30S70M", supp=false)
+    val r2Half2 = r("chr1", 120, Minus, r=2, cigar="30M70S", supp=true)
+
+    Seq(fullR1, r1Half1, r1Half2, fullR2, r2Half1, r2Half2).foreach { rec => rec.name = "q1" }
+
+    def test(recs: Seq[SamRecord], ids: Seq[String]): Unit = {
+      recs.length shouldBe ids.length
+      val input   = toInput(recs:_*)
+      val outputs = Outputs()
+      new SvPileup(input=input, output=outputs.prefix).execute()
+      val outRecs = readBamRecs(outputs.bam)
+      outRecs.length shouldBe recs.length
+      outRecs.zip(ids).foreach { case (rec, id) =>
+        withClue(rec.asSam.getSAMString.trim) {
+          rec.getOrElse[String](SvPileup.SamBreakpointTag, "") shouldBe id
+        }
+      }
+    }
+
+    val fromTag = s"0;from;${SplitRead.snakeName}"
+    val intoTag = f"0;into;${SplitRead.snakeName}"
+
+//    // r1Half2 is not annotated, since it is superseded by r1Half1 -> fullR2
+//    SamPairUtil.setMateInfo(r1Half1.asSam, fullR2.asSam, true)
+//    SamPairUtil.setMateInformationOnSupplementalAlignment(r1Half2.asSam, fullR2.asSam, true)
+//    test(Seq(r1Half2, r1Half1, fullR2), Seq(fromTag, "", intoTag))
```

Comment:
```suggestion
    // r1Half2 is not annotated, since it is superseded by r1Half1 -> fullR2
    SamPairUtil.setMateInfo(r1Half1.asSam, fullR2.asSam, true)
    SamPairUtil.setMateInformationOnSupplementalAlignment(r1Half2.asSam, fullR2.asSam, true)
    test(Seq(r1Half2, r1Half1, fullR2), Seq(fromTag, "", intoTag))
```